src/stats/overall.ts

```diff
@@ -112,10 +115,10 @@ function getBeneficialTradeRatio(conversionsByPlayerByOpening, playerIndex, oppo
   zippedTrades.forEach((conversionPair) => {
     const playerConversion = _.first(conversionPair);
     const opponentConversion = _.last(conversionPair);
-    const playerDamage = playerConversion.currentPercent - playerConversion.startPercent;
-    const opponentDamage = opponentConversion.currentPercent - opponentConversion.startPercent;
+    const playerDamage = (playerConversion as any).currentPercent - (playerConversion as any).startPercent;
+    const opponentDamage = (opponentConversion as any).currentPercent - (opponentConversion as any).startPercent;
 
-    if (playerConversion.didKill && !opponentConversion.didKill) {
+    if ((playerConversion as any).didKill && !(opponentConversion as any).didKill) {
```

Comment:
What is the purpose of as any here? Does it not understand the types of the objects? There should be types with the currentPercent and didKill fields right?

Reply:
It's because the argument `conversionsByPlayerByOpening` is declared as `any`. Can be solved by declaring it on both `getOpeningRatio()` and `getBeneficialTradeRatio()` as ` { [x: string]: _.Dictionary<ConversionType[]> }`

```ts
type ConversionDictionary = { [x: string]: _.Dictionary<ConversionType[]> };
// ...
function getOpeningRatio(conversionsByPlayerByOpening: ConversionDictionary, playerIndex: number, opponentIndex: number, type: any): Ratio {}
// ...
function getBeneficialTradeRatio(conversionsByPlayerByOpening: ConversionDictionary, playerIndex: number, opponentIndex: number): Ratio {}
// ...
const playerDamage = playerConversion.currentPercent - playerConversion.startPercent;
const opponentDamage = opponentConversion.currentPercent - opponentConversion.startPercent;

if (playerConversion.didKill && !opponentConversion.didKill) {
// ...
```

Reply:
Yeah ideally there are no `any`s in the Typescript code. I've fixed this particular instance of any, but there are some `any`s that remain in the codebase. This PR actually doesn't really change much of the fundamental code or the logic, but fixing the remaining `any`s would require a reassessment of the actual typings and logic behind them.

Reply:
I'm not a fan of defining the `ConversionDictionary` type because that's a result of one of `lodash`'s magical functions and is not really a type. It would be better to cast it as an `any` for now and then reassess how to use proper typings such as `Map<string, ConversionType>` to ensure strict typing in the future.